Support L10N Variable in make.conf to select only the Language will be installed
as it set in make.conf L10N
(befor it was LINGUAS)
correct some install path for libsi
correct install in the right order
signed-of-by: Joerg Bornkessel <hd_brummy@gentoo.org> (2018/Jan/21)
diff -Naur vdr-2.1.7.orig/Makefile vdr-2.1.7/Makefile
--- vdr-2.1.7.orig/Makefile	2015-01-18 12:37:45.506034673 +0100
+++ vdr-2.1.7/Makefile	2015-01-18 12:38:34.086037162 +0100
@@ -116,7 +116,7 @@
 VDRVERSION = $(shell sed -ne '/define VDRVERSION/s/^.*"\(.*\)".*$$/\1/p' config.h)
 APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' config.h)
 
-all: vdr i18n plugins
+all: vdr i18n
 
 # Implicit rules:
 
@@ -170,7 +170,7 @@
 
 PODIR     = po
 LOCALEDIR = locale
-I18Npo    = $(wildcard $(PODIR)/*.po)
+I18Npo    = $(foreach dir,$(L10N),$(wildcard $(PODIR)/$(dir)*.po))
 I18Nmo    = $(addsuffix .mo, $(foreach file, $(I18Npo), $(basename $(file))))
 I18Nmsgs  = $(addprefix $(LOCALEDIR)/, $(addsuffix /LC_MESSAGES/vdr.mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
 I18Npot   = $(PODIR)/vdr.pot
@@ -193,7 +193,7 @@
 
 install-i18n: i18n
 	@mkdir -p $(DESTDIR)$(LOCDIR)
-	cp -r $(LOCALEDIR)/* $(DESTDIR)$(LOCDIR)
+	@(cd $(LOCALEDIR); for l10n in $(L10N); do [ "$$l10n" = "en" ] && continue; cp -r --parents $$l10n* $(DESTDIR)$(LOCDIR); done)
 
 # The 'include' directory (for plugins):
 
@@ -255,7 +255,7 @@
 
 # Install the files (note that 'install-pc' must be first!):
 
-install: install-pc install-bin install-dirs install-conf install-doc install-plugins install-i18n install-includes
+install: install-pc install-bin install-dirs install-conf install-doc install-i18n install-includes
 
 # VDR binary:
 
@@ -267,12 +267,13 @@
 
 install-dirs:
 	@mkdir -p $(DESTDIR)$(VIDEODIR)
-	@mkdir -p $(DESTDIR)$(CONFDIR)
-	@mkdir -p $(DESTDIR)$(ARGSDIR)
-	@mkdir -p $(DESTDIR)$(CACHEDIR)
+#	@mkdir -p $(DESTDIR)$(CONFDIR)
+#	@mkdir -p $(DESTDIR)$(ARGSDIR)
+#	@mkdir -p $(DESTDIR)$(CACHEDIR)
 	@mkdir -p $(DESTDIR)$(RESDIR)
 
 install-conf:
+	@mkdir -p $(DESTDIR)$(CONFDIR)
 	@cp -pn *.conf $(DESTDIR)$(CONFDIR)
 
 # Documentation:
@@ -299,8 +300,11 @@
 # Includes:
 
 install-includes: include-dir
-	@mkdir -p $(DESTDIR)$(INCDIR)
-	@cp -pLR include/vdr include/libsi $(DESTDIR)$(INCDIR)
+#	@mkdir -p $(DESTDIR)$(INCDIR)
+#	@cp -pLR include/vdr include/libsi $(DESTDIR)$(INCDIR)
+	@mkdir -p $(DESTDIR)$(INCDIR)/vdr $(DESTDIR)$(INCDIR)/vdr/libsi
+	@cp -pLR include/vdr $(DESTDIR)$(INCDIR)
+	@cp -pLR include/libsi Make.config $(DESTDIR)$(INCDIR)/vdr
 
 # pkg-config file:
 
